### DISCUSSION TITLE
Error while build esp32 samples/hello_world

### DISCUSSION THREAD

**Discussion by Trafficode** [zephyrproject-rtos/zephyr#52643] [Q&A]
Hi
Following instruction(link below) of build simple example for esp32 I met some problem...
https://docs.zephyrproject.org/latest/boards/xtensa/esp32/doc/index.html

Version of zephyr project that I currently use:
VERSION_MAJOR = 3
VERSION_MINOR = 2
PATCHLEVEL = 99
VERSION_TWEAK = 0
EXTRAVERSION =

SDK version:
zephyr-sdk-0.15.2

Operationn system version:
Distributor ID:	Ubuntu
Description:	Ubuntu 22.04.1 LTS
Release:	22.04
Codename:	jammy

Build logs:
(.venv) kku@kku-CELSIUS:~/zephyrproject/zephyr$ west build -b esp32 samples/hello_world
[3/20] Performing configure step for 'EspIdfBootloader'
FAILED: esp-idf/src/EspIdfBootloader-stamp/EspIdfBootloader-configure /home/kku/zephyrproject/zephyr/build/esp-idf/src/EspIdfBootloader-stamp/EspIdfBootloader-configure 
cd /home/kku/zephyrproject/zephyr/build/esp-idf/build/bootloader && /usr/bin/cmake -GNinja -S /home/kku/zephyrproject/modules/hal/espressif/components/bootloader/subproject -B /home/kku/zephyrproject/zephyr/build/esp-idf/build/bootloader -DSDKCONFIG=/home/kku/zephyrproject/zephyr/build/esp-idf/build/sdkconfig -DIDF_PATH=/home/kku/zephyrproject/modules/hal/espressif -DIDF_TARGET=esp32 -DPYTHON_DEPS_CHECKED=1 -DCMAKE_C_COMPILER=/home/kku/zephyr-sdk-0.15.2/xtensa-espressif_esp32_zephyr-elf/bin/xtensa-espressif_esp32_zephyr-elf-gcc -DCMAKE_CXX_COMPILER=/home/kku/zephyr-sdk-0.15.2/xtensa-espressif_esp32_zephyr-elf/bin/xtensa-espressif_esp32_zephyr-elf-g++ -DCMAKE_ASM_COMPILER=/home/kku/zephyr-sdk-0.15.2/xtensa-espressif_esp32_zephyr-elf/bin/xtensa-espressif_esp32_zephyr-elf-gcc -DCMAKE_SYSTEM_NAME=Generic -DPYTHON=/home/kku/zephyrproject/.venv/bin/python3.10 && /usr/bin/cmake -E touch /home/kku/zephyrproject/zephyr/build/esp-idf/src/EspIdfBootloader-stamp/EspIdfBootloader-configure
CMake Warning at /home/kku/zephyrproject/modules/hal/espressif/tools/cmake/idf.cmake:18 (message):
  IDF_PATH environment variable is different from inferred IDF_PATH.

                              Check if your project's top-level CMakeLists.txt includes the right
                              CMake files. Environment IDF_PATH will be used for the build:
                              /home/kku/esp/ESP8266_RTOS_SDK
Call Stack (most recent call first):
  /home/kku/zephyrproject/modules/hal/espressif/tools/cmake/project.cmake:12 (include)
  CMakeLists.txt:52 (include)

CMake Error at /home/kku/zephyrproject/modules/hal/espressif/tools/cmake/idf.cmake:47 (include):
  include could not find requested file:

    dfu
Call Stack (most recent call first):
  /home/kku/zephyrproject/modules/hal/espressif/tools/cmake/project.cmake:12 (include)
  CMakeLists.txt:52 (include)

CMake Error at /home/kku/zephyrproject/modules/hal/espressif/tools/cmake/idf.cmake:48 (include):
  include could not find requested file:

    uf2
Call Stack (most recent call first):
  /home/kku/zephyrproject/modules/hal/espressif/tools/cmake/project.cmake:12 (include)
  CMakeLists.txt:52 (include)

-- Unexpected file in components directory: /home/kku/esp/ESP8266_RTOS_SDK/components/VERSION.md
-- Component directory /home/kku/esp/ESP8266_RTOS_SDK/components/VERSION.md does not contain a CMakeLists.txt file. No component will be added
-- The C compiler identification is unknown
-- The CXX compiler identification is unknown
-- The ASM compiler identification is unknown
-- Found assembler: xtensa-esp32-elf-gcc
CMake Error at /home/kku/zephyrproject/modules/hal/espressif/tools/cmake/project.cmake:290 (__project):
  The CMAKE_C_COMPILER:

    xtensa-esp32-elf-gcc

  is not a full path and was not found in the PATH.

  Tell CMake where to find the compiler by setting either the environment
  variable "CC" or the CMake cache entry CMAKE_C_COMPILER to the full path to
  the compiler, or to the compiler name if it is in the PATH.
Call Stack (most recent call first):
  CMakeLists.txt:59 (project)

CMake Error at /home/kku/zephyrproject/modules/hal/espressif/tools/cmake/project.cmake:290 (__project):
  The CMAKE_CXX_COMPILER:

    xtensa-esp32-elf-g++

  is not a full path and was not found in the PATH.

  Tell CMake where to find the compiler by setting either the environment
  variable "CXX" or the CMake cache entry CMAKE_CXX_COMPILER to the full path
  to the compiler, or to the compiler name if it is in the PATH.
Call Stack (most recent call first):
  CMakeLists.txt:59 (project)

CMake Error at /home/kku/zephyrproject/modules/hal/espressif/tools/cmake/project.cmake:290 (__project):
  The CMAKE_ASM_COMPILER:

    xtensa-esp32-elf-gcc

  is not a full path and was not found in the PATH.

  Tell CMake where to find the compiler by setting either the environment
  variable "ASM" or the CMake cache entry CMAKE_ASM_COMPILER to the full path
  to the compiler, or to the compiler name if it is in the PATH.
Call Stack (most recent call first):
  CMakeLists.txt:59 (project)

-- Warning: Did not find file Compiler/-ASM
-- Configuring incomplete, errors occurred!
See also "/home/kku/zephyrproject/zephyr/build/esp-idf/build/bootloader/CMakeFiles/CMakeOutput.log".
See also "/home/kku/zephyrproject/zephyr/build/esp-idf/build/bootloader/CMakeFiles/CMakeError.log".
ninja: build stopped: subcommand failed.
FATAL ERROR: command exited with status 1: /usr/bin/cmake --build /home/kku/zephyrproject/zephyr/build

**Comment by sylvioalves** [comment]
@Trafficode, have you `export ZEPHYR_TOOLCHAIN_VARIANT="zephyr"`?

**Reply by Trafficode** [reply]
Yeap, I can build the same example for diffrent board with no problems... Finally I think that I have a little bit mess on my ubuntu instalation. I tried the same on clean podman ubuntu 22.04 instalation and it works perfectly. In, past I was working with different sdk for esp8266 and I suppose thet there are some environment variables conflict... But I'm not sure, I didn't went deeper to this... I just work on podman instance :]
